### PR TITLE
feat: Add optional `keep_alive`

### PR DESCRIPTION
**Pull request by sentrivana**
In case folks are experiencing network issues between the SDK and Sentry, we in general recommend tweaking socket_options directly with a predefined set of settings. This seems to alleviate the issue most of the time, but feels rather low-level and hacky.

Node has a separate [keepAlive transport option](https://docs.sentry.io/platforms/node/configuration/options/#transport-options). We could do something similar: an option that sets the socket options to [what we usually recommend](https://github.com/getsentry/sentry-python/releases/tag/1.41.0), while keeping the `socket_options` option around as well for users who need more control over things.

Docs: https://github.com/getsentry/sentry-docs/pull/9498

Closes https://github.com/getsentry/sentry-python/issues/2837

---

## General Notes

Thank you for contributing to `sentry-python`!

Please add tests to validate your changes, and lint your code using `tox -e linters`.

Running the test suite on your PR might require maintainer approval. Some tests (AWS Lambda) additionally require a maintainer to add a special label to run and will fail if the label is not present.

#### For maintainers

Sensitive test suites require maintainer review to ensure that tests do not compromise our secrets. This review must be repeated after any code revisions.

Before running sensitive test suites, please carefully check the PR. Then, apply the `Trigger: tests using secrets` label. The label will be removed after any code changes to enforce our policy requiring maintainers to review all code revisions before running sensitive tests.

### CODE REVIEW

File: sentry_sdk/transport.py [hunk 1]
```diff
@@ -448,6 +464,8 @@ def _get_pool_options(self, ca_certs):
 
         if self.options["socket_options"]:
             options["socket_options"] = self.options["socket_options"]
+        elif self.options["keep_alive"]:
+            options["socket_options"] = KEEP_ALIVE_SOCKET_OPTIONS
```
Comment (antonpirker) [inline sentry_sdk/transport.py:468]: Wouldnt it be nice that people can set `keep_alive=True` but then also override some values in `socket_options`?

**Review by antonpirker — COMMENTED** [review]
I have a question.

File: sentry_sdk/transport.py (hunk 1 above)
Comment (sentrivana) [reply sentry_sdk/transport.py]: Added here: https://github.com/getsentry/sentry-python/pull/2842/commits/19534decf4512c7db132d6fd07c42c408a847930

File: tests/test_transport.py [hunk 2]
```diff
@@ -167,6 +167,58 @@ def test_socket_options(make_client):
...
+        (socket.SOL_TCP, socket.TCP_KEEPINTVL, 42),
+    ]
+
+    client = make_client(socket_options=socket_options, keep_alive=True)
+
+    options = client.transport._get_pool_options([])
+    assert options["socket_options"] == [
+        (socket.SOL_SOCKET, socket.SO_KEEPALIVE, 42),
+        (socket.SOL_TCP, socket.TCP_KEEPINTVL, 42),
+        (socket.SOL_TCP, socket.TCP_KEEPCNT, 6),
```
[the remark this answers is not in the document]
Comment (sentrivana) [reply tests/test_transport.py]: yup, fixed now

**Review by antonpirker — APPROVED** [review]
:shipit:

**Pull request merged**